feat(cli): add experimental LuaRocks wrapper using select

Vishrutha1909 opened the pull request:
This is an experimental LuaRocks wrapper UI, luarox.lua, located in the experimental/ directory. It provides a simple interactive interface for common LuaRocks commands using CLI widgets from the terminal.lua library.

Menu-driven interface with the following choices:
- luarocks install
- luarocks list
- luarocks search
- Exit

Internal prototype for testing and exploring interactive CLI patterns.

Tieske commented on experimental/luarox.lua:
```diff
@@ -0,0 +1,54 @@
+local function divider(title)
+  print("\n" .. ("-"):rep(40))
+  if title then print(title) end
+  print(("="):rep(40) .. "\n")
```
there is also `terminal.draw.line.title`

Tieske commented on src/terminal/utils.lua:
Can we not change this? Note the difference is that the table to the constructor becomes the object instance. See other comments.

Tieske commented on src/terminal/cli/select.lua:
```diff
@@ -0,0 +1,139 @@
+
+-- Initialize menu
+function Select:init(options)
+  assert(type(options) == "table", "options must be a table")
+  assert(type(options.choices) == "table", "choices must be a table")
+  assert(#options.choices > 0, "choices must not be empty")
+  for _, val in ipairs(options.choices) do
+    assert(type(val) == "string", "each choice must be a string")
+  end
```
```suggestion
  assert(type(self.choices) == "table", "choices must be a table")
  assert(#self.choices > 0, "choices must not be empty")
  for _, val in ipairs(self.choices) do
    assert(type(val) == "string", "each choice must be a string")
  end
```

Just replace `options` with `self` should work with the other class model.
Obviously other occurrences must also be adapted.

Tieske commented on src/terminal/cli/prompt.lua:
```diff
@@ -0,0 +1,20 @@
+
+function Prompt:run()
+  t.output.write(self.prompt .. " ")
+  return t.input.readansi(math.huge)
```
why does this work? it should return only a single key no?

Vishrutha1909 commented:
@Tieske Thanks for the feedback, Thijs!

I've updated the code as suggested:

- Replaced the manual divider() with draw.line.title() for consistency and better integration with the existing drawing utilities.
- Fixed argument handling for draw.line.title() - it now receives a plain string for the title and a numeric width from t.size().

As for what luarox currently does:

- It shows a Select menu with basic LuaRocks commands: install, list, search, and exit.
- If the user selects list, it runs luarocks list and shows the output.
- For install and search, it just prints confirmation for now ("You selected install/search").
- Exit cleanly quits the app.

I’ve temporarily left out the prompt-based input for install and search just to make sure the core flow works reliably ,especially since there was some confusion with terminal state and input handling earlier. I’ll reintroduce prompt support next now that the rest is stable.

Tieske commented:
Maybe we can use a temporary solution; the prompt would need the readline functionality. But for now we could use a temporary function that sets stdin back to blocking and canonical mode, then uses io.read, then reverts to non-blocking and disable canonical mode.

That function would then be blocking, but for now that's not an issue. Can be replaced later with readline.

Vishrutha1909 commented on src/terminal/utils.lua:
I changed it to make the minimal class model support passing a prefilled table as the instance. This was needed because in the CLI widgets like Select, I instantiate them with config directly: `local menu = Select{ prompt = "...", choices = {...} }` 
Without the change, the passed in table gets discarded and the init() method doesn’t receive the config. The new line added in utils.lua : `local instance = setmetatable(arg or {}, cls)` ensures that the user passed table becomes the instance which allows self.prompt to be directly set. Let me know if you'd prefer handling this differently.

Tieske replied:
sorry, I'm only seeing your comments now? missed that.

You are correct, in the implementation (in main currently) the options are not passed to `init`. That is because the options table passed into the constructor BECOMES the instance (by taking the opts table, and setting the class-metatable on it).

So where you do:

    self.some_prop = opts.some_prop or "default value"

You can change it to:

    self.some_prop = self.some_prop or "default_value"

because `self` is the `opts` table. See this PIL paragraph: https://www.lua.org/pil/16.1.html on OOP with Lua.

Tieske commented:
@Vishrutha1909 I just pushed a commit to your branch that implement the suggestions I made. Please have a look. I think this is good to go (we can squash the commits upon merging)

Vishrutha1909 commented:
> @Vishrutha1909 I just pushed a commit to your branch that implement the suggestions I made. Please have a look. I think this is good to go (we can squash the commits upon merging)

Thanks a lot for pushing the updates! I just reviewed the commit and everything looks good to me as well.
